javascript-bank/account.js

```diff
@@ -0,0 +1,48 @@
+/* exported Account */
+
+function Account(number, holder) {
+  this.number = number;
+  this.holder = holder;
+  this.transactions = [];
+}
+
+Account.prototype.deposit = function (amount) {
+  if (Number.isInteger(amount) !== true || amount < 1) {
+    return false;
+  } else {
+    var newDeposit = new Transaction();
+    newDeposit.type = 'deposit';
+    newDeposit.amount = amount;
```

Comment:
You should be passing these values as arguments to the Transaction function, not assigning the properties yourself on the return value.